cmd/plugin/credentialcomposer/dbcredentialcomposer/main.go

```diff
@@ -0,0 +1,144 @@
+// The server will supply the default attributes it will apply to the server X509-SVID. If the plugin returns an empty
+// response or NOT_IMPLEMENTED, the server will apply the default attributes. Otherwise, the returned attributes are
+// used. If an X509-SVID is produced that does not conform to the SPIFFE X509-SVID specification for leaf certificates,
+// it will be rejected. This function cannot be used to modify the SPIFFE ID of the X509-SVID.
+func (p *Plugin) ComposeServerX509SVID(ctx context.Context, req *credentialcomposerv1.ComposeServerX509SVIDRequest) (*credentialcomposerv1.ComposeServerX509SVIDResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+// ComposeAgentX509SVID implements the CredentialComposer ComposeAgentX509SVID RPC. Composes the SPIRE Agent X509-SVID.
+// The server will supply the default attributes it will apply to the agent X509-SVID. If the plugin returns an empty
+// response or NOT_IMPLEMENTED, the server will apply the default attributes. Otherwise, the returned attributes are used.
+// If an X509-SVID is produced that does not conform to the SPIFFE X509-SVID specification for leaf certificates, it will
+// be rejected. This function cannot be used to modify the SPIFFE ID of the X509-SVID.
+func (p *Plugin) ComposeAgentX509SVID(ctx context.Context, req *credentialcomposerv1.ComposeAgentX509SVIDRequest) (*credentialcomposerv1.ComposeAgentX509SVIDResponse, error) {
+	return nil, status.Error(codes.Unimplemented, "not implemented")
+}
+
+// ComposeWorkloadX509SVID implements the CredentialComposer ComposeWorkloadX509SVID RPC. Composes workload X509-SVIDs.
+// The server will supply the default attributes it will apply to the workload X509-SVID. If the plugin returns an empty
+// response or NOT_IMPLEMENTED, the server will apply the default attributes. Otherwise, the returned attributes are used.
+// If an X509-SVID is produced that does not conform to the SPIFFE X509-SVID specification for leaf certificates, it will
+// be rejected. This function cannot be used to modify the SPIFFE ID of the X509-SVID.
+func (p *Plugin) ComposeWorkloadX509SVID(ctx context.Context, req *credentialcomposerv1.ComposeWorkloadX509SVIDRequest) (*credentialcomposerv1.ComposeWorkloadX509SVIDResponse, error) {
+	config, err := p.getConfig()
+	if err != nil {
+		return nil, err
+	}
+
+	for _, mySQLPathPrefix := range config.MySQLSPIFFEIDPathPrefixes {
+		spiffeIDRaw := req.GetSpiffeId()
+		if strings.HasPrefix(spiffeIDRaw, mySQLPathPrefix) {
+			spiffeID, err := spiffeid.FromString(spiffeIDRaw)
+			if err != nil {
+				return nil, err
+			}
+
+			pathComponents := strings.Split(spiffeID.Path(), "/")
+			mySQLDBInstanceName := pathComponents[len(pathComponents)-1]
+			resp := &credentialcomposerv1.ComposeWorkloadX509SVIDResponse{
+				Attributes: &credentialcomposerv1.X509SVIDAttributes{
+					Subject: &credentialcomposerv1.DistinguishedName{
+						CommonName: mySQLDBInstanceName,
```

Comment:
what's the logic here for generating the `CommonName`? would be helpful to understand with an example

Reply:
I'll add a comment to clarify